src/generator/named-types.ts

```diff
@@ -97,25 +110,39 @@ function fetchNamedTypes(modules: Module[]): NamedTypesResult {
   const typeMap: Record<string, { namedType: NamedType; source: ValueTypeSource; associatedModules: Set<string> }> = {};
 
   modules.forEach((module) => {
-    fetchRootTypes(module).forEach(([valueType, source]) => {
-      recursiveVisitMembersType(valueType, (membersType, path) => {
-        let namedType = membersType;
-        if (isTupleType(namedType)) {
-          namedType = membersType as unknown as InterfaceType;
-          namedType.kind = ValueTypeKind.interfaceType;
-          namedType.name = path;
-          namedType.documentation = '';
-          namedType.customTags = {};
-        }
-
-        if (typeMap[namedType.name] === undefined) {
-          typeMap[namedType.name] = { namedType, source, associatedModules: new Set() };
-        }
-
-        const existingResult = typeMap[namedType.name];
-        existingResult.associatedModules.add(module.name);
-        existingResult.source |= source;
-      });
+    fetchRootTypes(module).forEach(({ valueType, source, uniquePath }) => {
+      recursiveVisitMembersType(
+        valueType,
+        (membersType, path) => {
+          let namedType = membersType as unknown as NamedType;
```

Comment:
Can we avoid doing this conversion? It is not necessary if you change the type of `namedType` in place, rather than defining `const enumType`.

Reply:
I change the membersType to NamedType is because UnionType has no property name and the compiler doesn't allow me to continue in line 136

Reply:
That's because you defined a new const `enumType` instead of changing the type of `namedType` in place like the tuple conversion does. If you do that, TypeScript compiler can correctly infer the type in line 136.

Reply:
I see, let me fix it